app/probe.go

```diff
@@ -0,0 +1,102 @@
+package app
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/arquivei/foundationkit/errors"
+)
+
+// Probe stores the state of a probe (`true` or `false` for ok and not ok respectively).
+type Probe struct {
+	ok *bool
+}
+
+// Set changes the state of a probe. Use `true` for ok and `false` for not ok.
+func (p *Probe) Set(ok bool) {
+	*p.ok = ok
+}
+
+// SetOk sets the probe as ok. Same as `Set(true)`.
+func (p *Probe) SetOk() {
+	*p.ok = true
+}
+
+// SetNotOk sets the probe as not ok. Same as `Set(false)`.
+func (p *Probe) SetNotOk() {
+	*p.ok = false
+}
+
+// IsOk returns the state of the probe  (`true` or `false` for ok and not ok respectively).
+func (p *Probe) IsOk() bool {
+	return *p.ok
+}
+
+// ProbeGroup aggregates and manages probes.
+type ProbeGroup struct {
+	lock   *sync.RWMutex
+	probes map[string]*bool
```

Comment:
This feels tricky.`probes` could be a `map[string]Probe` to avoid usage of pointers in both `ok *bool` and to store probes. I believe this makes intention clear as to why things are, or are not pointers. Pointers accept null values, but `nil` is not needed here imo.

Reply:
I used pointers because I didn't want to fetch a map item every time I changed the probe state. Even if I used a map o Probes instead of bools, I would have to have a pointer at some place.